econsa/tests/test_quantile_measures.py

```diff
@@ -0,0 +1,277 @@
+"""Tests for quantile based sensitivity measures.
+
+We implement tests replicating the results presented in Kucherenko et al. 2019.
+"""
+from functools import partial
+
+import numpy as np
+import pytest
+from numpy.testing import assert_array_almost_equal
+from scipy.stats import norm
+from temfpy.uncertainty_quantification import ishigami
+from temfpy.uncertainty_quantification import simple_linear_function
+
+from econsa.quantile_measures import mc_quantile_measures
+
+
+@pytest.fixture
+def test_1_fixture():
+    """First test case."""
+
+    # Objective function
+    def simple_linear_function_transposed(x):
+        """Simple linear function model but with variables stored in columns."""
+        return simple_linear_function(x.T)
+
+    # Analytical values of linear model with normally distributed variables
+    # are used as benchmarks for the first test case.
+    mean = np.array([1, 3, 5, 7])
+    cov = np.array(
+        [
+            [1, 0, 0, 0],
+            [0, 2.25, 0, 0],
+            [0, 0, 4, 0],
+            [0, 0, 0, 6.25],
+        ],
+    )
+    n_params = len(mean)
+
+    # range of alpha
+    dalp = (0.98 - 0.02) / 30
+    alp = np.arange(0.02, 0.98 + dalp, dalp)  # len(alp) = 31
+
+    # q_2: PDF of the out put Y(Eq.30)
+    q_2_true = [
+        (
+            cov[i, i]
+            + norm.ppf(a) ** 2
+            * (np.sqrt(np.trace(cov)) - np.sqrt(sum(cov[j, j] for j in range(n_params) if j != i)))
+            ** 2
+        )
+        for a in alp
+        for i in range(n_params)
+    ]
+    # reshape
+    q_2_true = np.vstack(q_2_true).reshape((len(alp), n_params))
+
+    # Q_2: normalized quantile based sensitivity measure 2.(Eq.14)
+    norm_q_2_true = [q[i] / sum(q) for q in q_2_true for i in range(n_params)]
+    norm_q_2_true = np.hstack(norm_q_2_true).reshape((len(alp), n_params))
+
+    out = {
+        "func": simple_linear_function_transposed,
+        "n_params": n_params,
+        "loc": mean,
+        "scale": cov,
+        "dist_type": "Normal",
+        "norm_q_2_true": norm_q_2_true,
+    }
+
+    return out
+
+
+def test_1(test_1_fixture):
+    norm_q_2_true = test_1_fixture["norm_q_2_true"]
+    func = test_1_fixture["func"]
+    n_params = test_1_fixture["n_params"]
+    loc = test_1_fixture["loc"]
+    scale = test_1_fixture["scale"]
+    dist_type = test_1_fixture["dist_type"]
+
+    for estimator, n_draws, decimal in zip(
+        ["DLR", "DLR", "DLR", "DLR", "brute force"],
+        [2 ** 6, 2 ** 9, 2 ** 10, 2 ** 13, 3000],
+        [0, 1, 1, 2, 2],
+    ):
+        norm_q_2_solve = mc_quantile_measures(
+            estimator=estimator,
+            func=func,
+            n_params=n_params,
+            loc=loc,
+            scale=scale,
+            dist_type=dist_type,
+            n_draws=n_draws,
+        )
```

Comment:
Did you try to use `partial` in more places? For example this looks like fit for me.